pkg/controller/ec2/vpcpeeringconnection/setup.go

```diff
@@ -0,0 +1,189 @@
+		Values: filterValue,
+	}
+	filters := []*svcsdk.Filter{&filter}
+	obj.SetFilters(filters)
+	return nil
+}
+
+func (e *custom) postObserve(_ context.Context, cr *svcapitypes.VPCPeeringConnection, obj *svcsdk.DescribeVpcPeeringConnectionsOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	connectionCounter := -1
+	for _, v := range obj.VpcPeeringConnections {
+		connectionCounter++
+		for _, tag := range v.Tags {
+			if *tag.Key == "Name" {
+				if *tag.Value == cr.ObjectMeta.Labels["peering-connection-id"] {
+					switch aws.StringValue(v.Status.Code) {
+					case string(svcapitypes.VPCPeeringConnectionStateReasonCode_pending_acceptance):
+						cr.SetConditions(xpv1.Creating())
+					case string(svcapitypes.VPCPeeringConnectionStateReasonCode_deleted):
+						cr.SetConditions(xpv1.Unavailable())
+						return managed.ExternalObservation{}, nil
+					case string(svcapitypes.VPCPeeringConnectionStateReasonCode_active):
+						cr.SetConditions(xpv1.Available())
+					}
+				}
+			}
+		}
+	}
+	cr.SetConditions(xpv1.Available())
+	return obs, nil
+}
+
+func (e *custom) isUpToDate(cr *svcapitypes.VPCPeeringConnection, obj *svcsdk.DescribeVpcPeeringConnectionsOutput) (bool, error) {
+	connectionCounter := -1
+
+	for _, v := range obj.VpcPeeringConnections {
+		connectionCounter++
+
+		for _, tag := range v.Tags {
+			if *tag.Key == "Name" {
+				if *tag.Value == cr.ObjectMeta.Name {
+					switch *v.Status.Code {
+					case "active":
+						return true, nil
+					case "deleted":
+						return false, nil
+					}
+
+					if *v.Status.Code == "pending-acceptance" && cr.Spec.ForProvider.AcceptRequest {
+						// if acceptRequest is true, we automatically accept the request on AWS
+						req := svcsdk.AcceptVpcPeeringConnectionInput{
+							VpcPeeringConnectionId: awsclients.String(*v.VpcPeeringConnectionId),
+						}
+
+						request, _ := e.client.AcceptVpcPeeringConnectionRequest(&req)
+						err := request.Send()
+						if err != nil {
+							return false, err
+						}
+						return true, nil
+					}
+				}
+			}
+		}
+	}
+	return false, nil
+}
+
+func preCreate(ctx context.Context, cr *svcapitypes.VPCPeeringConnection, obj *svcsdk.CreateVpcPeeringConnectionInput) error {
+	// set external name as tag on the vpc peering connection
+	resType := "vpc-peering-connection"
+	key := "Name"
+	value := meta.GetExternalName(cr)
+
+	spec := svcsdk.TagSpecification{
+		ResourceType: &resType,
+		Tags: []*svcsdk.Tag{
+			{
+				Key:   &key,
+				Value: &value,
+			},
+		},
+	}
+	obj.TagSpecifications = append(obj.TagSpecifications, &spec)
+
+	return nil
+}
+
+func (e *custom) postCreate(ctx context.Context, cr *svcapitypes.VPCPeeringConnection, obj *svcsdk.CreateVpcPeeringConnectionOutput, cre managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	// set peering connection id as label on the k8s object after creation
```

Comment:
Why do we need it as a label? Doesn't it exist as external-name annotation already?

Reply:
No, external-name is the metadata.name from the claim but peering connection id is generated dynamically by aws. But we might also want to have the peering connection id available in the cluster, so we set it as a label

Reply:
Why not use the connection id as external name, similar to [other resources](https://github.com/crossplane/provider-aws/blob/7f29cf3/pkg/controller/apigatewayv2/apimapping/setup.go#L90) whose id is generated after creation?